src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -1386,8 +1387,8 @@ def load_arguments(self, _):
         c.argument('disk_encryption_set_name', disk_encryption_set_name)
         c.argument('key_url', help='URL pointing to a key or secret in KeyVault.')
         c.argument('source_vault', help='Name or ID of the KeyVault containing the key or secret.')
-        c.argument('encryption_type', arg_type=get_enum_type(['EncryptionAtRestWithPlatformKey', 'EncryptionAtRestWithCustomerKey', 'EncryptionAtRestWithPlatformAndCustomerKeys']),
-                   help='The type of key used to encrypt the data of the disk. EncryptionAtRestWithPlatformKey: Disk is encrypted at rest with Platform managed key. It is the default encryption type. EncryptionAtRestWithCustomerKey: Disk is encrypted at rest with Customer managed key that can be changed and revoked by a customer. EncryptionAtRestWithPlatformAndCustomerKeys: Disk is encrypted at rest with 2 layers of encryption. One of the keys is Customer managed and the other key is Platform managed.')
+        c.argument('encryption_type', arg_type=get_enum_type(['EncryptionAtRestWithPlatformKey', 'EncryptionAtRestWithCustomerKey', 'EncryptionAtRestWithPlatformAndCustomerKeys', 'ConfidentialVmEncryptedWithCustomerKey']),
```

Comment:
We should be able to get these valid values from the `DiskEncryptionSetType` enumeration in Python SDK, right?

Reply:
Could you modify the Swagger to include `ConfidentialVmEncryptedWithCustomerKey` in the latest SDK package?

Reply:
@zhoxing-ms - Please refer to https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskEncryptionSet.json Swagger desfinition, I can see **ConfidentialVmEncryptedWithCustomerKey** listed on line 463

Reply:
@AjKundnani OK, why is `EncryptionAtRestWithPlatformKey` not defined in the Swagger?

Reply:
@anshulsolanki21 Could you define two independent enumerations for these two different scenarios?

Reply:
OK, I think we can remove `EncryptionAtRestWithPlatformKey` for `--encryption-type` when ignite Event or when some users request an issue, because removing it directly will cause breaking change on the CLI side **(even if the breaking change has occurred on the server side)**

Reply:
If I understand correctly, CLI implements both enum with all 3 values, however DiskRP would return bad request if 'EncryptionAtRestWithPlatformKey' is specified for diskencryptiontype
You mean "we can remove EncryptionAtRestWithPlatformKey for --encryption-type" for diskencryptiontype, right?

sure, I would let you decide when to remove this value

Reply:
Sure, thank you for your understanding！